### DISCUSSION TITLE
Anyone tried to use multiple electron populations in Thomson?

### DISCUSSION THREAD

**Discussion by vicente-valenzuela** [PlasmaPy/PlasmaPy#2905] [Analysis & Diagnostics]
Hello all,

I'm working on a plasmapy-based Thomson scattering pipeline for laser and pulsed-power experiments. It is straight-forward to use multiple ion populations (thanks to great documentation!), but with electrons plasmapy seems to break. In particular, the Thomson package breaks when it calculates the plasma wavenumbers "ks" and "kl" (lines 406 and 407). The error reads "ValueError: operands could not be broadcast together with shapes (243,) (2,) "

Does anyone have suggestions?

Thanks!
Vicente

**Comment by namurphy** [comment]
@pheuer — Might you have any suggestions here?  Thanks!

**Comment by pheuer** [comment]
@VicenteValenzuelaV Can you post an minimal example of inputs that causes this error? It's true that we don't have multiple electron populations in the tests - I will open a PR to add them, but when I did the test passes, so I think it's working? 

Also, the documentation notebook still seems to be running with two electron species: https://docs.plasmapy.org/en/stable/notebooks/diagnostics/thomson.html 

Edit: PR with test https://github.com/PlasmaPy/PlasmaPy/pull/2906
